frontend/test/metabase/scenarios/question/notebook.cy.spec.js

```diff
@@ -388,6 +388,76 @@ describe("scenarios > question > notebook", () => {
         );
       });
     });
+
+    it.skip("should join saved question with sorted metric (metabase#13744)", () => {
+      cy.server();
+      // create first question based on repro steps in #13744
+      withSampleDataset(({ PRODUCTS }) => {
+        cy.request("POST", "/api/card", {
+          name: "13744",
+          dataset_query: {
+            database: 1,
+            query: {
+              "source-table": 1,
+              aggregation: [["count"]],
+              breakout: [["field-id", PRODUCTS.CATEGORY]],
+              "order-by": [["asc", ["aggregation", 0]]],
+            },
+            type: "query",
+          },
+          display: "table",
+          visualization_settings: {},
+        }).then(({ body: { id: questionId } }) => {
+          const ALIAS = `Question ${questionId}`;
+
+          // create new question and join it with a previous one
+          cy.request("POST", "/api/card", {
+            name: "13744_joined",
+            dataset_query: {
+              database: 1,
+              query: {
+                joins: [
+                  {
+                    alias: ALIAS,
+                    fields: "all",
+                    condition: [
+                      "=",
+                      ["field-id", PRODUCTS.CATEGORY],
+                      [
+                        "joined-field",
+                        ALIAS,
+                        ["field-literal", "CATEGORY", "type/Text"],
+                      ],
+                    ],
+                    "source-table": `card__${questionId}`,
+                  },
+                ],
+                "source-table": 1,
+              },
+              type: "query",
+            },
+            display: "table",
+            visualization_settings: {},
+          }).then(({ body: { id: joinedQuestionId } }) => {
+            cy.route("POST", `/api/card/${joinedQuestionId}/query`).as(
+              "cardQuery",
+            );
+
+            // Assert phase begins here
+            cy.visit(`question/${joinedQuestionId}`);
```

Comment:
Not sure if I understand what is being done here.
So `route` is just a listener, and still requires viewing the question visually with `visit`?
Also the `visit` doesn't start with `/`.

Reply:
Instead of waiting for loading spinner to disappear (relying on UI which is unreliable) in order to determine when the question **actually** loaded, I am relying on XHR.
If you check the console, whenever you load ANY question the very last XHR that fires is always `POST /api/card/:card-id/query`. That's a fail-safe way to know that data loaded.

Good catch about `visit` not starting with `/`. I'll fix that in a second.

Reply:
@flamber 
For example:
```javascript
cy.get(".LoadingSpinner").should("not.exist");
cy.findByText("There was a problem with your question").should("not.exist");
// and even when I add:
cy.findByText("Showing 0 rows").should("not.exist");
```
This was still passing locally - giving me false positive. It's simply too risky to rely on UI.

On the other hand:
```javascript
cy.wait("@cardQuery").then(xhr => {
  expect(xhr.response.body.error).not.to.exist;
});
// if I had put UI error check after XHR (which is redundant), it would **always** consistently fail like it should
cy.findByText("There was a problem with your question").should("not.exist");
```
If there is nothing wrong with the question, `response.body` will never have `error` in it. That's the first check that we can use to see if everything is ok.
And only after that I can use UI for **positive** assertions. There is a TODO note that I should add positive assertions once this issue is fixed. Or, if you know at least one value that would be in the UI I can add it right now.

Reply:
Okay, so it is listening to a request instead of visually.
The positive assertion would be any of these - first line is fine - (I just removed the sorting, then the query works)
![image](https://user-images.githubusercontent.com/1447303/98821694-2ef88080-2430-11eb-8b26-2a4d3b8b3f4e.png)